Client/Frontend/Browser/BrowserViewController/BrowserViewController+Onboarding.swift

```diff
@@ -98,94 +116,117 @@ extension BrowserViewController {
+      popover.arrowDistance = 10.0
+      
+      // Create a border / placeholder view
+      let borderView = BorderView(frame: frame, cornerRadius: cornerRadius, colouredBorder: true)
+      let placeholderView = UIView(frame: frame).then {
+        $0.alpha = 0.0
+        $0.frame = frame
       }
+      
+      view.addSubview(placeholderView)
+      popover.view.insertSubview(borderView, aboveSubview: popover.view)
 
       let maskShape = CAShapeLayer().then {
         $0.fillRule = .evenOdd
         $0.fillColor = UIColor.white.cgColor
         $0.strokeColor = UIColor.clear.cgColor
       }
+      
+      popover.present(from: placeholderView, on: self) { [weak popover, weak self] in
+        guard let popover = popover,
+          let self = self
+        else { return }
+
+        // Mask the shadow
+        let maskFrame = self.view.convert(frame, to: popover.backgroundOverlayView)
+        guard !maskFrame.isNull &&
+              !maskFrame.isInfinite &&
+              !maskFrame.isEmpty &&
+              !popover.backgroundOverlayView.bounds.isNull &&
+              !popover.backgroundOverlayView.bounds.isInfinite &&
+              !popover.backgroundOverlayView.bounds.isEmpty else {
+          return
+        }
 
+        guard maskFrame.origin.x.isFinite &&
+              maskFrame.origin.y.isFinite &&
+              maskFrame.size.width.isFinite &&
+              maskFrame.size.height.isFinite &&
+              maskFrame.size.width > 0 &&
+              maskFrame.size.height > 0 else {
+          return
+        }
+      }
+    
       popover.backgroundOverlayView.layer.mask = maskShape
+      
       popover.popoverDidDismiss = { _ in
         maskShape.removeFromSuperlayer()
         borderView.removeFromSuperview()
+
+        didDismiss()
       }
 
-      controller.buttonTapped = { [weak self] in
+      borderView.didClickBorderedArea = {
         maskShape.removeFromSuperlayer()
         borderView.removeFromSuperview()
-        DispatchQueue.main.async {
-          self?.openPrivacyReport()
+        
+        popover.dismissPopover() {
+          didClickBorderedArea()
         }
       }
-
+      
       DispatchQueue.main.async {
         maskShape.path = {
           let path = CGMutablePath()
           path.addRect(popover.backgroundOverlayView.bounds)
-          path.addRoundedRect(
-            in: maskFrame,
-            cornerWidth: 12.0,
-            cornerHeight: 12.0)
           return path
         }()
       }
-    }
   }
-
-  func notifyTrackersBlocked(domain: String, trackerName: String, remainingTrackersCount: Int) {
+  
+  func notifyTrackersBlocked(domain: String, displayTrackers: [AdBlockTrackerType], trackerCount: Int) {
     let controller = WelcomeBraveBlockedAdsController().then {
-      $0.setData(domain: domain, trackerBlocked: trackerName, trackerCount: remainingTrackersCount)
+      $0.setData(displayTrackers: displayTrackers.map { $0.rawValue }, trackerCount: trackerCount)
```

Comment:
Nitpick: This can use key path here `map(\.rawValue)` 🙂

Reply:
Fix at https://github.com/brave/brave-ios/pull/5302/commits/2947dc932af097abe91af16dde7f053847bf1303